voxblox/include/voxblox/test/layer_test_utils.h

```diff
@@ -76,6 +76,37 @@ class LayerTest {
   static constexpr double kTolerance = 1e-10;
 };
 
+template <typename VoxelType>
+void LayerTest<VoxelType>::CompareVoxel(const VoxelType& voxel_A,
+                                        const VoxelType& voxel_B) const {
+  LOG(FATAL) << "Not implemented for this voxel type!";
+}
+
+template <>
+void LayerTest<EsdfVoxel>::CompareVoxel(const EsdfVoxel& voxel_A,
+                                        const EsdfVoxel& voxel_B) const {
+  constexpr double kTolerance = 1e-10;
+
+  CHECK_NEAR(voxel_A.distance, voxel_B.distance, kTolerance);
```

Comment:
Don't use checks in tests!!! Use ASSERT for stuff that should fail the test and can not go on, use EXPECT for stuff that should fail the test but the tests below can be run without crashing the program.